File: client-clojure/src/sirqul_io_t_platform/specs/user_permissions.clj
(ns sirqul-io-t-platform.specs.user-permissions
  (:require [clojure.spec.alpha :as s]
            [spec-tools.data-spec :as ds]
            [sirqul-io-t-platform.specs.connection-group :refer :all]
            [sirqul-io-t-platform.specs.connection :refer :all]
            [sirqul-io-t-platform.specs.permissions :refer :all]
            [sirqul-io-t-platform.specs.account :refer :all]
            )
  (:import (java.io File)))


(def user-permissions-data
  {
   (ds/opt :id) int?
   (ds/opt :active) boolean?
   (ds/opt :valid) boolean?
   (ds/opt :permissionableType) string?
   (ds/opt :permissionableId) int?
   (ds/opt :connectionGroup) connection-group-spec
   (ds/opt :friendGroup) boolean?
   (ds/opt :connection) connection-spec
   (ds/opt :permissions) permissions-spec
   (ds/opt :exclude) boolean?
   (ds/opt :blocked) boolean?
   (ds/opt :owner) account-spec
   })

(def user-permissions-spec
  (ds/spec
    {:name ::user-permissions
     :spec user-permissions-data}))

File: client-scalaz/src/main/scala/org/openapitools/client/api/UserPermissions.scala
package org.openapitools.client.api

import argonaut._
import argonaut.EncodeJson._
import argonaut.DecodeJson._

import org.http4s.{EntityDecoder, EntityEncoder}
import org.http4s.argonaut._
import org.joda.time.DateTime


import UserPermissions._

case class UserPermissions (
  id: Option[Long],
active: Option[Boolean],
valid: Option[Boolean],
permissionableType: Option[String],
permissionableId: Option[Long],
connectionGroup: Option[ConnectionGroup],
friendGroup: Option[Boolean],
connection: Option[Connection],
permissions: Option[Permissions],
exclude: Option[Boolean],
blocked: Option[Boolean],
owner: Option[Account])

object UserPermissions {
  import DateTimeCodecs._

  implicit val UserPermissionsCodecJson: CodecJson[UserPermissions] = CodecJson.derive[UserPermissions]
  implicit val UserPermissionsDecoder: EntityDecoder[UserPermissions] = jsonOf[UserPermissions]
  implicit val UserPermissionsEncoder: EntityEncoder[UserPermissions] = jsonEncoderOf[UserPermissions]
}
